lib/widgets/photo_filter.dart

```diff
@@ -53,207 +55,361 @@ class PhotoFilter extends StatelessWidget {
 
 ///The PhotoFilterSelector Widget for apply filter from a selected set of filters
 class PhotoFilterSelector extends StatefulWidget {
-  final Widget title;
+  const PhotoFilterSelector(
+      {Key key,
+      @required this.title,
+      @required this.filters,
+      @required this.image,
+      this.appBarColor = Colors.blue,
+      this.loader = const Center(child: CircularProgressIndicator()),
+      this.fit = BoxFit.fill,
+      @required this.filename,
+      this.selectFilter,
+      this.circleShape = false})
+      : super(key: key);
+
   final Color appBarColor;
+  final bool circleShape;
+  final String filename;
   final List<Filter> filters;
-  final imageLib.Image image;
-  final Widget loader;
   final BoxFit fit;
-  final String filename;
-  final bool circleShape;
-
-  const PhotoFilterSelector({
-    Key key,
-    @required this.title,
-    @required this.filters,
-    @required this.image,
-    this.appBarColor = Colors.blue,
-    this.loader = const Center(child: CircularProgressIndicator()),
-    this.fit = BoxFit.fill,
-    @required this.filename,
-    this.circleShape = false,
-  }) : super(key: key);
+  final imagelib.Image image;
+  final Widget loader;
+  final Filter selectFilter;
+  final Widget title;
 
   @override
-  State<StatefulWidget> createState() => new _PhotoFilterSelectorState();
+  State<StatefulWidget> createState() => _PhotoFilterSelectorState();
 }
 
 class _PhotoFilterSelectorState extends State<PhotoFilterSelector> {
+  Map<String, List<int>> cachedImageFilters = {};
+  Map<String, List<int>> cachedThumbnailFilters = {};
   String filename;
-  Map<String, List<int>> cachedFilters = {};
-  Filter _filter;
-  imageLib.Image image;
+  imagelib.Image image; //original image
   bool loading;
+  imagelib.Image thumbnailImage; // thumbnail image
+
+  Filter _filter;
+
+  @override
+  void dispose() {
+    if (!Executor().isClosed) {
+      Executor().close();
+    }
+    super.dispose();
+  }
 
   @override
   void initState() {
     super.initState();
-    loading = false;
-    _filter = widget.filters[0];
+    loading = Executor().isClosed;
+    warmUpExecutor();

```

Comment:
If isolates pools start in previous activity/widget execution will be fast. Basically this warmUpExector & Executor().close in initState & dispose of previous statefulwidget respectively.